yetanotherco/aligned_layer
# remove: emit NewBatchV2 from Aligned.sol

# This PR
This PR removes `emit NewBatchV2` from `AlignedLayerServiceManager.sol`, but not removed from the interface to keep bindings compatible, as discussed.

I have also pushed the new bindings, so you can see how only a small fraction of bytecode changes, and the functions to read `NewBatchV2` emitions is still there.

## To Deploy
This PR changes `AlignedLayerServiceManager.sol` , so it should be redeployed in v0.7.0

## To Test

(same as #1025 )

### Retrocompatibility

1) Deploy all infra on `testnet` except Operator
2) Deploy Operator on this branch, ON ANOTHER REPO
- Don't forget to set `var switchBlockNumber = uint32(0)` in `operator.go`, `aggregator.go` and `subscriber.go` instances
- Also set `if block.number < 0`in `AlignedLayerServiceManager.sol`
3) Verify Operator receives new batches and verifies them correctly
- Note: check operator responds task only once, this is because only one event is emitted.

### Normal Compatibility 
1) Deploy all the network on this branch, and test normal flow
